### PR TITLE
Add support for system theme

### PR DESCRIPTION
**Pull request by moraj-turing**
This PR implements system theme feature suggested on  #978

### CODE REVIEW

**Comment by cuu508** [comment]
Perhaps we can use `light-dark` function when it has more browser support, and use the JS solution in the meantime.

File: templates/base.html [hunk 1]
```diff
@@ -233,6 +233,29 @@
...
     {% endcompress %}
     {% endblock %}
+    {% if request.user.is_authenticated and request.profile.theme == 'system' %}
+    <script>
+        // Check to see if Media-Queries are supported
+        if (window.matchMedia) {
+            // Check if the dark-mode Media-Query matches
+            if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+                // Create a new meta tag element
+                var metaTag = $('<meta>');
```
Comment (cuu508) [inline templates/base.html:243]: The meta tag does not appear to have any function, can it be removed?

File: templates/base.html [hunk 2]
```diff
@@ -233,6 +233,29 @@
...
+
+                // Set attributes for the meta tag
+                metaTag.attr('name', 'description');
+                metaTag.attr('content', 'Your meta tag content goes here');
+
+                // Append the meta tag to the head element
+                $('head').append(metaTag);
+                $("body").addClass("dark");
+                console.log('dark mode enabled');
+            } else {
```
Comment (cuu508) [inline templates/base.html:253]: The else branch does nothing and can be removed.

File: templates/base.html [hunk 3]
```diff
@@ -233,6 +233,29 @@
...
+                var metaTag = $('<meta>');
+
+                // Set attributes for the meta tag
+                metaTag.attr('name', 'description');
+                metaTag.attr('content', 'Your meta tag content goes here');
+
+                // Append the meta tag to the head element
+                $('head').append(metaTag);
+                $("body").addClass("dark");
+                console.log('dark mode enabled');
```
Comment (cuu508) [inline templates/base.html:252]: Please remove the debug line

File: templates/base.html [hunk 4]
```diff
@@ -233,6 +233,29 @@
     <script src="{% static 'js/bootstrap.min.js' %}"></script>
     {% endcompress %}
     {% endblock %}
+    {% if request.user.is_authenticated and request.profile.theme == 'system' %}
+    <script>
+        // Check to see if Media-Queries are supported
+        if (window.matchMedia) {
+            // Check if the dark-mode Media-Query matches
+            if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (cuu508) [inline templates/base.html:241]: The two ifs can be combined into one.

**Comment by moraj-turing** [comment]
Updated with comments

File: templates/base.html [hunk 5]
```diff
@@ -233,6 +230,24 @@
     <script src="{% static 'js/bootstrap.min.js' %}"></script>
     {% endcompress %}
     {% endblock %}
+    {% if request.user.is_authenticated and request.profile.theme == 'system' %}
+    <script>
+        // Check to see if Media-Queries are supported
+        if (window.matchMedia?.('(prefers-color-scheme: dark)').matches) {
+            // Create a new meta tag element
+            var metaTag = $('<meta>');
```
Comment (cuu508) [inline templates/base.html:238]: Does the meta tag have any function? Can it be removed?

File: templates/base.html [hunk 6]
```diff
@@ -233,6 +230,24 @@
...
+
+            // Set attributes for the meta tag
+            metaTag.attr('name', 'description');
+            metaTag.attr('content', 'Your meta tag content goes here');
+
+            // Append the meta tag to the head element
+            $('head').append(metaTag);
+            $("body").addClass("dark");
+            console.log('dark mode enabled');
```
Comment (cuu508) [inline templates/base.html:247]: Please remove the console.log line

**Comment by moraj-turing** [comment]
Hey @cuu508 I have updated the PR with your comments

**Comment by cuu508** [comment]
@moraj-turing I still see the meta tag and the `console.log` line in the PR.

**Pull request merged**